pkg/controllers/federatedresourcequota/federated_resource_quota_enforcement_controller.go

```diff
@@ -0,0 +1,219 @@
+			if apierrors.IsNotFound(err) {
+				err := c.collectQuotaStatuses(ctx, binding.Namespace)
+				if err != nil {
+					return controllerruntime.Result{}, err
+				}
+			} else {
+				return controllerruntime.Result{}, err
+			}
+		}
+	} else {
+		if !quota.DeletionTimestamp.IsZero() {
+			return controllerruntime.Result{}, nil
+		}
+		if err := c.collectQuotaStatus(quota, false); err != nil {
+			klog.Errorf("Failed to collect status for FederatedResourceQuota(%s), error: %v", req.NamespacedName.String(), err)
+			c.EventRecorder.Eventf(quota, corev1.EventTypeWarning, events.EventReasonCollectFederatedResourceQuotaStatusFailed, err.Error())
+			return controllerruntime.Result{}, err
+		}
+	}
+
+	c.EventRecorder.Eventf(quota, corev1.EventTypeNormal, events.EventReasonCollectFederatedResourceQuotaStatusSucceed, "Collect status of FederatedResourceQuota(%s) succeed.", req.NamespacedName.String())
+	return controllerruntime.Result{}, nil
+}
+
+// SetupWithManager creates a controller and register to controller manager.
+func (c *QuotaEnforcementController) SetupWithManager(mgr controllerruntime.Manager) error {
+	fn := handler.MapFunc(
+		func(ctx context.Context, obj client.Object) []reconcile.Request {
+			var quotaList policyv1alpha1.FederatedResourceQuotaList
+			if err := c.Client.List(ctx, &quotaList, client.InNamespace(obj.GetNamespace())); err != nil {
+				return nil
+			}
+
+			if len(quotaList.Items) == 0 {
+				return nil // no quotas in namespace, skip reconcile
+			}
+
+			return []reconcile.Request{{
+				NamespacedName: types.NamespacedName{
+					Namespace: obj.GetNamespace(),
+					Name:      obj.GetName(),
+				},
+			}}
+		},
+	)
+
+	resourceBindingPredicate := builder.WithPredicates(predicate.Funcs{
+		CreateFunc: func(_ event.CreateEvent) bool {
+			return false // ignore creates
+		},
+		UpdateFunc: func(_ event.UpdateEvent) bool {
+			return false // ignore updates
+		},
+		DeleteFunc: func(_ event.DeleteEvent) bool {
+			return true // only care about RB deletions
+		},
+		GenericFunc: func(event.GenericEvent) bool {
+			return false
+		},
+	})
+	return controllerruntime.NewControllerManagedBy(mgr).
+		For(&policyv1alpha1.FederatedResourceQuota{}).
+		Watches(&workv1alpha2.ResourceBinding{}, handler.EnqueueRequestsFromMapFunc(fn), resourceBindingPredicate).
+		Complete(c)
+}
+
+func (c *QuotaEnforcementController) collectQuotaStatuses(ctx context.Context, namespace string) error {
+	var quotaList policyv1alpha1.FederatedResourceQuotaList
+	if err := c.Client.List(ctx, &quotaList, client.InNamespace(namespace)); err != nil {
+		return err
+	}
+
+	for _, quota := range quotaList.Items {
+		err := c.collectQuotaStatus(&quota, true)
+		if err != nil {
+			fmt.Printf("failed to collect status for FederatedResourceQuota(%s), error: %v", klog.KObj(&quota).String(), err)
+		}
+	}
+
+	return nil
+}
+
+func (c *QuotaEnforcementController) collectQuotaStatus(quota *policyv1alpha1.FederatedResourceQuota, fullUpdate bool) error {
```

Comment:
Refer to https://github.com/karmada-io/karmada/pull/6367#discussion_r2088331745, no need to introduce parameter `fullUpdate` to distinguish between quota and resourceBinding.
```suggestion
func (c *QuotaEnforcementController) collectQuotaStatus(quota *policyv1alpha1.FederatedResourceQuota) error {
```